### PR TITLE
PC-708: Adjust weekly feedback download date range to one clear week

### PR DESCRIPTION
**Pull request by chidin194**
There are some formatting changes included here that aren't related to my changes but have appeared after running `make check-python-code`

### CODE REVIEW

File: help_to_heat/portal/download_views.py [hunk 1]
```diff
@@ -181,22 +183,16 @@ def download_feedback_view(request):
...
-    referrals = portal_models.Referral.objects.filter(created_at__gte=datetime.now() - timedelta(days=7)).order_by("referral_id")
-    downloaded_at = timezone.now()
-    file_name = downloaded_at.strftime("%d-%m-%Y %H_%M")
+    today = date.today()
+    end_of_last_week = today - timedelta(days=today.weekday())
+    start_of_last_week = end_of_last_week - timedelta(days=7)
+    referrals = portal_models.Referral.objects.filter(
+        created_at__gte=start_of_last_week, created_at__lt=end_of_last_week
+    ).order_by("referral_id")
+    file_name = f"Weekly Referrals ({datetime.strptime(str(start_of_last_week), '%Y-%m-%d').strftime('%d-%m-%Y')})"
```
Comment (chidin194) [inline help_to_heat/portal/download_views.py:195]: This is perhaps a little messy, but it seems there's no method for just formatting a date, so it seems we have to parse the date start_of_last_week as a string to a datetime, and then format this back into a string with just the date.

**Comment by jamiehumphries** [comment]
> There are some formatting changes included here that aren't related to my changes but have appeared after running `make check-python-code`

Thanks for clarifying @chidin194.

No need to change this time, but in future it would be helpful if such changes could be separated out into a separate pull request, or at least a separate commit within this one.

Separating them out makes a clear distinction between functional code changes and style ones.

File: help_to_heat/portal/download_views.py (hunk 1 above)
Comment (jamiehumphries) [reply help_to_heat/portal/download_views.py:195]: Are you sure about this? I was able to just run:
```
start_of_last_week.strftime("%d-%m-%Y")
```

Here it is in the docs:
https://docs.python.org/3/library/datetime.html#datetime.date.strftime

> `date.strftime(format)`
Return a string representing the date, controlled by an explicit format string. Format codes referring to hours, minutes or seconds will see 0 values. See also [strftime() and strptime() Behavior](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior) and [date.isoformat()](https://docs.python.org/3/library/datetime.html#datetime.date.isoformat).

File: help_to_heat/portal/download_views.py [hunk 2]
```diff
@@ -181,22 +183,16 @@ def download_feedback_view(request):
...
-    return response
-
-
 @require_http_methods(["GET"])
 @decorators.requires_service_manager
 def download_referrals_last_week_view(request):
-    referrals = portal_models.Referral.objects.filter(created_at__gte=datetime.now() - timedelta(days=7)).order_by("referral_id")
-    downloaded_at = timezone.now()
-    file_name = downloaded_at.strftime("%d-%m-%Y %H_%M")
+    today = date.today()
```
Comment (jamiehumphries) [inline help_to_heat/portal/download_views.py:189]: Please can you add a comment to explain the boundaries, like:
```
# Weekly boundaries are Mondays at midnight (00:00).
today = …
end_of_last_week = …
start_of_last_week = …
```

It's not immediately obvious from the code, so good to help out future devs.

File: help_to_heat/portal/download_views.py [hunk 3]
```diff
@@ -181,22 +183,16 @@ def download_feedback_view(request):
...
+    end_of_last_week = today - timedelta(days=today.weekday())
+    start_of_last_week = end_of_last_week - timedelta(days=7)
```
Comment (jamiehumphries) [inline help_to_heat/portal/download_views.py:191]: I think we can actually use:
```
- timedelta(weeks=1)
```
https://docs.python.org/3/library/datetime.html#timedelta-objects

**Review by jamiehumphries — CHANGES_REQUESTED** [review]

File: help_to_heat/portal/download_views.py (hunk 1 above)
Comment (chidin194) [reply help_to_heat/portal/download_views.py]: Ah, thank you! It didn't seem quite right. I've changed this 😄

File: help_to_heat/portal/download_views.py (hunk 2 above)
Comment (chidin194) [reply help_to_heat/portal/download_views.py:190]: Added!

File: help_to_heat/portal/download_views.py (hunk 3 above)
Comment (chidin194) [reply help_to_heat/portal/download_views.py]: Changed this! Is the advantage here increased readability?

Comment (jamiehumphries) [reply help_to_heat/portal/download_views.py]: Yes, just a style thing. We may as well use it, since it's available.

**Review by jamiehumphries — APPROVED** [review]
Looks good! Let's go ahead and merge.

**Pull request merged**